pkg/controllers/nodepool/hash/suite_test.go

```diff
@@ -134,4 +139,124 @@ var _ = Describe("Static Drift Hash", func() {
 
 		Expect(nodePool.Annotations).To(HaveKeyWithValue(v1beta1.NodePoolHashAnnotationKey, expectedHash))
 	})
+	It("should update nodepool hash version when the nodepool hash version is out of sync with the controller", func() {
+		nodePool.Annotations[v1beta1.NodePoolHashVersionAnnotationKey] = "test"
+		ExpectApplied(ctx, env.Client, nodePool)
+		nodePool = ExpectExists(ctx, env.Client, nodePool)
+
+		Expect(nodePool.Annotations).To(HaveKeyWithValue(v1beta1.NodePoolHashAnnotationKey, "123456"))
```

Comment:
Do we need these expectation functions to assert startup conditions for our test case?